server/src-lib/Hasura/SQL/DML.hs

```diff
@@ -460,6 +460,7 @@ data BoolExp
   | BENotNull !SQLExp
   | BEExists !Select
   | BEIN !SQLExp ![SQLExp]
+  | BEFunc !T.Text ![SQLExp]
```

Comment:
Why not have `BEExp !SQLExp` so that any SQLExp of type boolean can be  used? Avoids having to define `SEFnApp` like constructor for `BoolExp`